src/factories/playhead.ts

```diff
@@ -0,0 +1,66 @@
+import { Container } from 'pixi.js'
+import { rectangleFactory } from '@/factories/rectangle'
+import { waitForApplication, waitForViewport } from '@/objects'
+import { waitForConfig } from '@/objects/config'
+import { emitter } from '@/objects/events'
+import { waitForScale } from '@/objects/scale'
+
+const autoViewportUpdatePadding = 80
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export async function playheadFactory() {
+  const application = await waitForApplication()
+  const viewport = await waitForViewport()
+  const config = await waitForConfig()
+  let scale = await waitForScale()
+  let initialized = false
+
+  const element = new Container()
+
+  const playhead = await rectangleFactory()
+  element.addChild(playhead)
+
+  emitter.on('scaleUpdated', updated => scale = updated)
+
+  const ticker = (): void => {
+    render()
+    checkViewport()
+  }
+
+  function render(): void {
+    playhead.width = config.styles.playheadWidth
+    playhead.height = application.stage.height
+    playhead.tint = config.styles.playheadColor
+    element.position.x = scale(new Date()) * viewport.scale._x + viewport.worldTransform.tx
```

Comment:
why `viewport.scale._x` and not `viewport.scale.x`?

Reply:
Not sure tbh, grabbed from v1 but `.x` seems to work just fine so updated it.